clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -118,8 +118,8 @@ public Selector(int maxReceiveSize, long connectionMaxIdleMs, Metrics metrics, T
         this.metricsPerConnection = metricsPerConnection;
     }
 
-    public Selector(long connectionMaxIdleMS, Metrics metrics, Time time, String metricGrpPrefix, Map<String, String> metricTags, ChannelBuilder channelBuilder) {
-        this(NetworkReceive.UNLIMITED, connectionMaxIdleMS, metrics, time, metricGrpPrefix, metricTags, true, channelBuilder);
+    public Selector(long connectionMaxIdleMS, Metrics metrics, Time time, String metricGrpPrefix, ChannelBuilder channelBuilder) {
+        this(NetworkReceive.UNLIMITED, connectionMaxIdleMS, metrics, time, metricGrpPrefix, metrics.config().tags(), true, channelBuilder);
```

Comment:
Got it. The reason that we added the tags to the sensor is the following (quote from the RB in KAFKA-1928).

"Gwen Shapira 6 months, 1 week ago (May 27, 2015, 8 p.m.)
I ran into an issue when I tried using a single Metrics object:

I initiated a single Metrics object and passed it into all the Selectors. 
Each selector had its own "processor-id" tags. When the Selectors initialize, each Sensor is re-used by all Selectors. The different MetricsNames with the different processor-id tags are all added with to the same Sensor. When we call Sensor.record - all statistics are updated, for all Selectors. Which means that all Selectors have the same statistics, which cannot be correct.

I solved this with multiple Metrics objects. Did I miss a better solution?
Jun Rao 6 months, 1 week ago (May 27, 2015, 8:40 p.m.)
This seems to be an issue with the Seletor. Basically, when creating a sensor in a selector, we didn't include the metricTags in the sensor name. Therefore, the sensors in the selector will have the same name if the metric instance is the same. We can fix this by including the tags in the sensor name. The sensor name just have to be unique and is not tied to the external jmx name. So, we can probably just add strings like "tagKey-tagValue" to the sensor name. This will make the construction of the sensor name more expensive. We can probably create a separate jira to optimize that a bit (e.g. by caching the global and the per node sensor names)."

So, including the customized tags explicitly specified through the constructor of Selector is enough to make the sensor names unique. We don't have to include the default tag in the sensor name. Could you verify that on the broker the jmx stats on different selectors in SocketServer are still correct w/o including the default tag in the sensor name?